test/compiler.test.js

```diff
@@ -1195,6 +1339,63 @@ describe('compiler', function() {
         }
       });
     });
+
+    it('loads component relative to appRootDir', function() {
+      appdir.writeConfigFileSync('./component-config.json', {
+        './component': { }
+      });
+      var appJS = appdir.writeConfigFileSync('component.js', '');
+
+      var instructions = boot.compile(appdir.PATH);
+      expect(instructions.components[0]).have.property(
+        'sourceFile', appJS
+      );
+    });
+
+    it('loads component relative to node modules', function() {
+      appdir.writeConfigFileSync('component-config.json', {
+        'index': { }
+      });
+      var appJS = appdir.writeConfigFileSync('node_modules/index.js', '');
+
+      var instructions = boot.compile(appdir.PATH);
+      expect(instructions.components[0]).have.property(
+        'sourceFile', appJS
+      );
+    });
+
+    it('prefers coffeescript over json for relative path component',
+      function() {
+      appdir.writeConfigFileSync('component-config.json', {
+        './component': { }
+      });
+
+      var coffee = appdir.writeFileSync('component.coffee', '');
+      appdir.writeFileSync('component.json', '');
+
+      var instructions = boot.compile(appdir.PATH);
+
+      expect(instructions.components).to.have.length(1);
+      expect(instructions.components[0]).have.property(
+        'sourceFile', coffee);
+    });
+
+    it('prefers coffeescript over json for non-relative path component',
```

Comment:
Please rename to "... for module relative component path" to keep this consistent with other test names.